core/services/eth/pool.go

```diff
@@ -43,9 +61,14 @@ func (p *Pool) Dial(ctx context.Context) (err error) {
 	return p.verifyChainIDs(ctx)
 }
 
+// verifyChainIDs checks that every node's chain ID is consistent, initializing from the first node if nil.
 func (p *Pool) verifyChainIDs(ctx context.Context) (err error) {
 	if p.chainID == nil {
-		return nil
+		chainID, err2 := p.nodes[0].ChainID(ctx)
```

Comment:
I actually think we can move verify into the node's Dial function now since you fixed the tests.

I'll handle that in my PR.